fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -800,6 +902,9 @@ private void addIndexCommon(@Nonnull Index index) {
         if (index.getAddedVersion() <= 0) {
             index.setAddedVersion(index.getLastModifiedVersion());
         }
+        if (useCounterBasedSubspaceKey && !index.hasExplicitSubspaceKey()) {
+            index.setSubspaceKey(++indexSubspaceKeyCounter);
+        }
```

Comment:
I don't think this attempts to make an effort not to assign subspace keys that overlap with explicit subspace keys, which maybe isn't the end of the world. (If there are clashes, it should be caught during the validation stage.) But it does seem to me like maybe checking if an explicit subspace key is a long and, if so, bumping the counter to the max of its current value and the explicit key might be nice to avoid unnecessary clashes. Or something like that.